ci/steps/publish-npm.sh

```diff
@@ -47,22 +47,25 @@ main() {
     exit 1
   fi
 
-  # We need TAG to know what to publish under on npm
-  # Options are "latest", "beta", or "<pr number >"
-  # See Environment comments above to know when each is used.
-  if ! is_env_var_set "NPM_TAG"; then
-    echo "NPM_TAG is not set. This is needed for tagging the npm release."
+  # We use this to grab the PR_NUMBER
+  if ! is_env_var_set "GITHUB_REF"; then
+    echo "GITHUB_REF is not set. Are you running this locally? We rely on values provided by GitHub."
     exit 1
   fi
 
-  echo "using tag: $NPM_TAG"
+  # We use this when setting NPM_VERSION
+  if ! is_env_var_set "GITHUB_SHA"; then
+    echo "GITHUB_SHA is not set. Are you running this locally? We rely on values provided by GitHub."
+    exit 1
+  fi
 
   # This allows us to publish to npm in CI workflows
   if [[ ${CI-} ]]; then
     echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
   fi
 
-  download_artifact npm-package ./release-npm-package
+  # NOTE@jsjoeio - this script assumes we have the artifact downloaded on disk
+  # That happens in CI as a step before we run this.
```

Comment:
This is awesome, I always felt the download step should be done as a CI step instead.

I see the download action for the beta publish job, should there be a call to `download_artifact` in the npm/brew workflow as well?  (Or maybe I have missed it.)

Reply:
Excellent catch! I thought I added it but it slipped through. You're right, it should be here: https://github.com/coder/code-server/blob/21bc065c204aae1cebe7c0b5d1c6f4b3c71ffcea/.github/workflows/npm-brew.yaml#L19-L20